externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/kyuubi/SparkSQLEngineEventListener.scala

```diff
@@ -50,6 +51,13 @@ class SparkSQLEngineEventListener(
     event match {
       case e: SessionEvent => updateSessionStore(e)
       case e: SparkOperationEvent => updateStatementStore(e)
+      case e: SparkListenerEvent if SparkUtilsHelper.lineageClassesArePresent =>
+        e match {
+          case OperationLineageEventWrapper(lineageEvent) =>
+            updateStatementLineageStore(lineageEvent)
```

Comment:
Here we can collect both `operationEvent` and `lineageEvent`, whether to consider adding `statementId` into the `operationEvent` with the same `executionId`

Reply:
> Here we can collect both `operationEvent` and `lineageEvent`, whether to consider adding `statementId` into the `operationEvent` with the same `executionId`

If we want to get the statementId we may need to use SQLOperationListener and make larger change, should we do that?

Reply:
How about we get the `statementId` via `qe.sparkSession.sparkContext.getLocalProperty("kyuubi.statement.id")` in `SparkOperationLineageQueryExecutionListener#onSuccess`?

Reply:
`SparkOperationEvent` was been stored in the same `kvstore` with `LineageEvent `, can we associate them here?

The listener is a separate thread from the execution thread of the spark session,  and I'm not sure if we can get it correctly.

Reply:
The KVStore seems to require statementId to get SparkOperationEvent.

Reply:
You are right, it seems to be difficult to correlate the two.